superset/jinja_context.py

```diff
@@ -240,7 +240,39 @@ def _schema_table(
             schema, table_name = table_name.split(".")
         return table_name, schema
 
-    def latest_partition(self, table_name: str):
+    def latest_partition(self, table_name: str) -> str:
+        """
+        Supports the backwards compatible Jinja template API from before
+        https://github.com/apache/incubator-superset/pull/7782. Returns the first
+        (or only) value in the array of all latest partitions for the given table name.
+
+        :param table_name: table name in the format `schema.table`
+        :return: the first (or only) value in the latest partition array
+        """
+
+        return self._first_latest_partition(table_name)
+
+    def _first_latest_partition(self, table_name: str) -> str:
```

Comment:
Do we need this function as it's only called once from above. Additionally the logic looks similar to `latest_partitions` and thus it seems line #253 could simply be, 

```
return self.latest_partitions(table_name)[0]
``` 

Note for both `latest_partition` and `latest_partitions` an `IndexError` could be thrown and thus it's probably prudent to add mention in the docstring, i.e., 

```
:param table_name: table name in the format `schema.table`
:return: the first (or only) value in the latest partition array
:raises IndexError: If no partition exists
```

Reply:
updated